src/performance/gpu/asus/asus_wmi.rs

```diff
@@ -31,7 +40,24 @@ impl AsusWmi {
 
     /// Returns the currently set STAPM value
     pub async fn tdp(&self) -> TDPResult<f64> {
-        Ok(self.platform.get_ppt_pl1_spl()? as f64)
+        if let Some(firmware_attributes) = &self.attributes {
+            match firmware_attributes
+                .attributes()
+                .iter()
+                .find(|a| a.name() == "ppt_pl1_spl")
+            {
+                Some(attribute) => match attribute.current_value() {
+                    Ok(attr_value) => match attr_value {
+                        AttrValue::Integer(value) => Ok(value as f64),
+                        _ => Err(TDPError::FailedOperation("Failed to read SPL.".to_string())),
+                    },
+                    Err(_) => Err(TDPError::FailedOperation("Failed to read SPL.".to_string())),
+                },
+                None => Err(TDPError::FailedOperation("Failed to read SPL.".to_string())),
+            }
+        } else {
+            Ok(self.platform.get_ppt_pl1_spl()? as f64)
+        }
```

Comment:
You can use `let Some(firmware_attributes) = ... else {}` pattern to reduce the amount of nesting here:

```suggestion
        let Some(firmware_attributes) = &self.attributes else {
            return Ok(self.platform.get_ppt_pl1_spl()? as f64);
        };
        match firmware_attributes
            .attributes()
            .iter()
            .find(|a| a.name() == "ppt_pl1_spl")
        {
            Some(attribute) => match attribute.current_value() {
                Ok(attr_value) => match attr_value {
                    AttrValue::Integer(value) => Ok(value as f64),
                    _ => Err(TDPError::FailedOperation("Failed to read SPL.".to_string())),
                },
                Err(_) => Err(TDPError::FailedOperation("Failed to read SPL.".to_string())),
            },
            None => Err(TDPError::FailedOperation("Failed to read SPL.".to_string())),
        }
```